File: StkExtensionPlugins/Hpop.ForceModels/CSharp/Srp/IExample1.cs
//=====================================================//
//  Copyright 2005, Analytical Graphics, Inc.          //
//=====================================================//
using System;

// NOTE: Indicate that your Interface for your plugin is within
// the same namespace as your plugin.
namespace AGI.Hpop.Plugin.Examples.Stk.ForceModeling.Srp.CSharp
{
	// NOTE:  Name your custom COM Interface with the identical name as your
	// plugin class's name, and append an I to the beginning of it.
	public interface IExample1
	{
		// NOTE:  Add your custom COM Interface Property configuration settings here.
		//        Follow the standard C# rules for exposing properties.
		string	Name					{ get; set; }
		bool	Enabled					{ get; set; }
		bool	DebugMode				{ get; set; }
		int		MsgInterval				{ get; set; }
		double	SpecularReflectivity	{ get; set; }
		double	DiffuseReflectivity		{ get; set; }
	}
}
//=====================================================//
//  Copyright 2005, Analytical Graphics, Inc.          //
//=====================================================//

File: StkExtensionPlugins/Hpop.ForceModels/VB.NET/Srp/IExample1.vb
'//=====================================================//
'//  Copyright 2005, Analytical Graphics, Inc.          //
'//=====================================================//

Imports System.Array

' NOTE:  Name your custom COM Interface with the identical name as your
' plugin class's name, and append an I to the beginning of it.
Public Interface IExample1

    ' NOTE:  Add your custom COM Interface Property configuration settings here.
    '        Follow the standard VB.NET rules for exposing properties.
    Property MyName() As String
    Property Enabled() As Boolean
    Property DebugMode() As Boolean
    Property MsgInterval() As Integer
    Property SpecularReflectivity() As Double
    Property DiffuseReflectivity() As Double

End Interface

'//=====================================================//
'//  Copyright 2005, Analytical Graphics, Inc.          //
'//=====================================================//